docs/retrieval-augmented-generation/rag-repo.md

```diff
@@ -0,0 +1,110 @@
+# New repository for RAG
+
+| Created  | Dec 5, 2024 |
+| -------- | -------- |
+| Authors | Bill Murdock |
+| Replaces | N/A |
+| Replaced by | N/A |
+
+## What
+
+We want a retrieval-augmented generation (RAG) capability that provides outstanding results with minimal effort, is seamlessly integrated with InstructLab, and is also general enough to be used in other applications as well.
+
+## Why
+
+Many InstructLab users want to train a model and then use it to RAG.  Often they build something simple themselves for this purpose.  Two problems with this approach:
+
+- Building their own RAG is extra work.
+- Users who are not experts on RAG might not build a RAG that provides outstanding results.
+
+There is a very simple RAG capability at <https://github.com/instructlab/rag> .  It is not tightly integrated with InstructLab and it does not use any advanced RAG capabilities.  However, it might have some existing users so we can't just unilaterally delete it or replace it with something radically different.
+
+## Goals
+
+Provide a built-in alternative for users who do not want to build their own RAG.  Do not break the existing capability at <https://github.com/instructlab/rag> .
+
+## Non-goals
+
+Evaluation of RAG will be addressed in one or more other development documents.  That topic is out of scope for this document.
+
+## Decision
+
+- There will be a new repository for RAG.
+- It will be located at <https://github.com/instructlab/retrieve>
+- By mid-January, it will be available and working but not integrated with InstructLab.
+- By mid-March, it will be integrated with InstructLab with the new repository being invoked by the core repository and maybe also by the SDG repository.
+- Eventually, it will be integrated with InstructLab with the new repository being invoked only by the core repository.
```

Comment:
I'm not sure I agree with necessarily assigning dates to these efforts from now. I think it does make sense to split up these efforts into phases, and then we can internally scope out work for the desired releases.

Reply:
I think the dates are important, but I am fine with dropping them from this document and using other venues to get aligned around these dates.  I will do that.